R/github-pat.R

```diff
@@ -1,40 +1,65 @@
+#' available, and securely prompts the user for credentials when needed.
 #'
 #' Packages that require a `GITHUB_PAT` can call this function to automatically
 #' set the `GITHUB_PAT` when needed. Users may call this function in their
 #' [.Rprofile][Startup] script to automatically set `GITHUB_PAT` for each R
 #' session without hardcoding any tokens on disk in plain-text.
 #'
 #' @export
-#' @param force_new forget existing pat, always ask for new one.
-#' @param validate checks with the github API that this token works. Defaults to
-#' `TRUE` only in an interactive R session (not when running e.g. CMD check).
-#' @param verbose prints a message showing the credential helper and PAT owner.
-#' @return Returns `TRUE` if a valid GITHUB_PAT was set, and FALSE if not.
-set_github_pat <- function(force_new = FALSE, validate = interactive(), verbose = validate){
+#'
+#' @param url A URL that conveys login information, such as protocol (e.g.
+#'   "https"), remote hostname (e.g. "github.com" or
+#'   "github.starkindustries.com"), and username (e.g. "tony"). See the
+#'   input/output documentation for the [`git-credential`
+#'   command](https://git-scm.com/docs/git-credential#IOFMT) for acceptable
+#'   formats.
+#' @param force_new Clear any existing, matching PAT from the store and
+#'   re-prompt the user.
+#' @param validate Whether to query the `/user` endpoint of the GitHub API to
+#'   confirm that the token works. Defaults to `TRUE` only in an interactive R
+#'   session (not when running e.g. CMD check). This API call respects a host
+#'   that is not "github.com", i.e. it works for GitHub Enterprise.
+#' @param verbose Whether to print a message revealing, e.g., the host, the PAT
+#'   owner's username, and the associated credential helper.
+#'
+#' @return Returns `TRUE` if `GITHUB_PAT` was set, and FALSE otherwise.
+#'
+#' @examples
+#' \dontrun{
+#' set_github_pat()
+#'
+#' set_github_pat("https://github.starkindustries.com")
+#'
+#' set_github_pat("https://tony@github.starkindustries.com")
+#' }
+set_github_pat <- function(url = "https://github.com",
+                           force_new = FALSE,
+                           validate = interactive(),
+                           verbose = validate){
+  # TODO: insert GITHUB_PAT_USER, if defined
   if(isTRUE(force_new))
-    git_credential_forget('https://token@github.com')
+    git_credential_forget(url)
   if(isTRUE(verbose))
     message("If prompted for GitHub credentials, enter your PAT in the password field")
   askpass <- Sys.getenv('GIT_ASKPASS')
   if(nchar(askpass)){
     # Hack to override prompt sentence to say "Token" instead of "Password"
     Sys.setenv(GIT_ASKTOKEN = askpass)
     Sys.setenv(GIT_ASKPASS = system.file('ask_token.sh', package = 'credentials', mustWork = TRUE))
-    Sys.setenv(GIT_ASKTOKEN_NAME = 'Personal Access Token (PAT)')
+    Sys.setenv(
+      GIT_ASKTOKEN_NAME = sprintf('%s Personal Access Token (PAT)', parse_url(url)["host"])
+    )
```

Comment:
It seems nice to remind people of the `host`, especially because, in indirect scenarios, the user might not have explicitly provided the `url`.